### PR TITLE
[extractor/xfileshare] Add Referer in HTTP header

### PR DESCRIPTION
**Pull request by Galiley**
<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### Description of your *pull request* and other information

</details>

DESCRIPTION
Add Referer in HTTP header

### CODE REVIEW

**Comment by pukkandan** [comment]
Why?

**Comment by Galiley** [comment]
Uqload needs the referer http headers to be set in order to be able to download the video.
I should have restricted it to only Uqload...

**Comment by pukkandan** [comment]
no open issue, no explanation, or even an example URL... how do you expect me to merge a PR like that?

Add a test case to the extractor. Or, if an existing test covers this already, tell mw which one

File: yt_dlp/extractor/xfileshare.py [hunk 1]
```diff
@@ -68,6 +68,17 @@ class XFileShareIE(InfoExtractor):
...
+        'info_dict': {
+            'id': 'nn85ef9zmgzk',
+            'ext': 'mp4',
+            'title': r're:.*',
+            'thumbnail': r're:https://.*\.jpg',
+            'http_headers': {
+                'Referer': 'https://uqload.com/nn85ef9zmgzk'
+            }
+        }
+    }, {
```
Comment (pukkandan) [inline yt_dlp/extractor/xfileshare.py:81]: 1. Use non-piracy link
2. Add md5
3. Use actual title
4. `http_headers` is not needed in test

**Review by pukkandan — CHANGES_REQUESTED** [review]
Does this cause any issues?

```diff
        return {
            'id': video_id,
            'title': title,
            'thumbnail': thumbnail,
            'formats': formats,
+           'http_headers': {'Referer': url},
        }
```

File: yt_dlp/extractor/xfileshare.py [hunk 2]
```diff
@@ -177,10 +186,7 @@ def _real_extract(self, url):
                         entry_protocol='m3u8_native', m3u8_id='hls',
                         fatal=False))
                 else:
-                    formats.append({
-                        'url': video_url,
-                        'format_id': 'sd',
-                    })
+                    formats.append({'url': video_url, 'format_id': 'sd'})
```
Comment (pukkandan) [inline yt_dlp/extractor/xfileshare.py:189]:
```suggestion
                    formats.append({
                        'url': video_url,
                        'format_id': 'sd',
                    })
```

**Pull request merged**